rules/terraformrules/terraform_blacklisted_resources.go

```diff
@@ -0,0 +1,77 @@
+package terraformrules
+
+import (
+	"fmt"
+	"log"
+	"sort"
+
+	"github.com/terraform-linters/tflint/tflint"
+)
+
+// TerraformBlacklistedResourcesRule checks whether variables have a type declared
+type TerraformBlacklistedResourcesRule struct{}
+
+type terraformBlacklistedResourcesRuleConfig struct {
+	Types map[string]string `hcl:"types"`
+}
+
+// NewTerraformBlacklistedResourcesRule returns a new rule
+func NewTerraformBlacklistedResourcesRule() *TerraformBlacklistedResourcesRule {
+	return &TerraformBlacklistedResourcesRule{}
+}
+
+// Name returns the rule name
+func (r *TerraformBlacklistedResourcesRule) Name() string {
+	return "terraform_blacklisted_resources"
+}
+
+// Enabled returns whether the rule is enabled by default
+func (r *TerraformBlacklistedResourcesRule) Enabled() bool {
+	return false
+}
+
+// Severity returns the rule severity
+func (r *TerraformBlacklistedResourcesRule) Severity() string {
+	return tflint.WARNING
+}
+
+// Link returns the rule reference link
+func (r *TerraformBlacklistedResourcesRule) Link() string {
+	return tflint.ReferenceLink(r.Name())
+}
+
+// Check whether resources exist of a ceratin type and provides a custom message
+func (r *TerraformBlacklistedResourcesRule) Check(runner *tflint.Runner) error {
+	log.Printf("[TRACE] Check `%s` rule for `%s` runner", r.Name(), runner.TFConfigPath())
+
+	config := terraformBlacklistedResourcesRuleConfig{}
+	config.Types = make(map[string]string)
+
+	if err := runner.DecodeRuleConfig(r.Name(), &config); err != nil {
+		return err
+	}
+
+	// Since the resources are stored as a map, there is no guarantee of the order in which
```

Comment:
This seems like something we should address in `AssertIssues` rather than requiring a rule to sort like this

Reply:
I agree, I just could not see a way to make this happen easily. The right thing to do is to resolve the issue in `AssertIssues` first instead of polluting the rule.

Terraform stores the resources as a `map`, so there is no guaranteed order in the case of evaluation of resources. 
```
type Issue struct {
	Rule    Rule
	Message string
	Range   hcl.Range
	Callers []hcl.Range
}
```

Thoughts?